src/ctr_decode_msgpack.c

```diff
@@ -117,6 +117,12 @@ static int unpack_instrumentation_scope_attributes(mpack_reader_t *reader, size_
         result = unpack_cfl_kvlist(reader, &attributes->kv);
 
         if (result != 0) {
+            if (attributes != NULL) {
```

Comment:
You are right about the leak but please use `ctr_attributes_destroy` on `attributes` so if that object is modified in the future we're covered.

Additionally, would you mind adding a `NULL` check in line 112? It seems like I missed it and the result of `ctr_attributes_create` is not being checked which sucks (my bad!).

If you add the `NULL` check in line 112 you won't need to check if `attributes` is `NULL` here because `unpack_cfl_kvlist` only modifies the pointer held by the `kv` member.

Thanks a lot!